guides/source/active_storage_overview.md

```diff
@@ -0,0 +1,397 @@
+**DO NOT READ THIS FILE ON GITHUB, GUIDES ARE PUBLISHED ON http://guides.rubyonrails.org.**
+
+Active Storage
+==============
+
+This guide covers how to attach files to your Active Record models.
+
+After reading this guide, you will know:
+
+* How to attach one or many files to a record.
+* How to delete an attached file.
+* How to link to an attached file.
+* How to use variants to transform images.
+* How to generate an image representation of a non-image file, such as a PDF or a video.
+* How to send file uploads directly from browsers to a storage service,
+  bypassing your application servers.
+* How to implement support for additional storage services.
+* How to clean up files stored during testing.
+
+--------------------------------------------------------------------------------
+
+What is Active Storage?
+-----------------------
+
+Active Storage facilitates uploading files to a cloud storage service like
+Amazon S3, Google Cloud Storage, or Microsoft Azure Storage and attaching those
+files to Active Record objects. It comes with a local disk-based service for
+development and testing and supports mirroring files to subordinate services for
+backups and migrations.
+
+Using Active Storage, an application can transform image uploads with
+[ImageMagick](https://www.imagemagick.org), generate image representations of
+non-image uploads like PDFs and videos, and extract metadata from arbitrary
+files.
+
+## Setup
+
+To setup an existing application after upgrading to Rails 5.2, run `rails
+active_storage:install`. If you're creating a new project with Rails 5.2,
+ActiveStorage will be installed by default. Installation generates a migration
+to add the tables needed to store attachments.
+
+If you wish to transform your images, add `mini_magick` to your Gemfile:
+
+``` ruby
+gem 'mini_magick'
+```
+
+Inside a Rails application, you can set up your services through the generated
+`config/storage.yml` file and reference one of the supported service types under
+the `service` key.
```

Comment:
Elaborate on the structure of `config/storage.yml` and provide an example:

> Declare Active Storage services in `config/storage.yml`. For each service your application uses, provide a name and the requisite configuration. The example below declares three services named `local`, `test`, and `s3`:
>
>```yaml
> local:
>   service: Disk
>   root: <%= Rails.root.join("storage") %>
>
> test:
>   service: Disk
>   root: <%= Rails.root.join("tmp/storage") %>
>
> s3:
>   service: S3
>   access_key_id: ""
>   secret_access_key: ""
> ```
>
> Continue reading for information on the built-in service adapters (e.g. `Disk` and `S3`) and the configuration they require.

Expand on how and when to specify which service Active Storage should use:

> Tell Active Storage which service to use by setting `Rails.application.config.active_storage.service`. Because each environment will likely use a different service, it is recommended to do this on a per-environment basis. To use the `disk` service from the previous example in the development environment, you would add the following to `config/environments/development.rb`:
>
> ```ruby
> # Store files locally.
> config.active_storage.service = :disk
> ```
>
> To use the `s3` service in production, you would add the following to `config/environments/production.rb`:
>
> ```ruby
> # Store files in S3.
> config.active_storage.service = :s3
> ```

Because populating `config/storage.yml` and setting `Rails.application.config.active_storage.service` are required and using image variants is optional, I would put the information on adding `mini_magick` to the Gemfile last.